contracts/governance/Vesting/fouryear/FourYearVestingLogic.sol

```diff
@@ -137,6 +138,42 @@ contract FourYearVestingLogic is IFourYearVesting, FourYearVestingStorage, Appro
 		emit DividendsCollected(msg.sender, _loanPoolToken, _receiver, _maxCheckpoints);
 	}
 
+	/**
+	 * @notice Change token owner - only vesting owner.
+	 * @dev Modifies token owner. This must be followed by approval
+	 * from token owner.
+	 * @param _newTokenOwner Address of new token owner.
+	 * */
+	function changeTokenOwner(address _newTokenOwner) public onlyOwner {
+		require(_newTokenOwner != address(0), "invalid new token owner address");
+		require(_newTokenOwner != tokenOwner, "same owner not allowed");
+		newTokenOwner = _newTokenOwner;
+	}
+
+	/**
+	 * @notice Approve token owner change - only token Owner.
+	 * @dev Token owner can only be modified
+	 * when both vesting owner and token owner have approved. This
+	 * function ascertains the approval of token owner.
+	 * */
+	function approveOwnershipTransfer() public onlyTokenOwner {
+		require(newTokenOwner != address(0), "invalid address");
+		tokenOwner = newTokenOwner;
+		newTokenOwner = address(0);
+		emit TokenOwnerChanged(tokenOwner, msg.sender);
+	}
+
+	/**
+	 * @notice Set address of the implementation - only Token Owner.
+	 * @dev This function sets the new implementation address.
+	 * It must also be approved by the Vesting owner.
+	 * @param _newImplementation Address of the new implementation.
+	 * */
+	function setImpl(address _newImplementation) public onlyTokenOwner {
```

Comment:
why implementation can be changed by the tokenOwner only?
form the task (story) statement:
```
Important:  both exchequer and investor need to confirm a contract update. exchequer could be the owner and the invesor the token owner.
```

Reply:
1. setImpl() can only be called by the token owner. This function doesn't change implementation. It just modifies a state variable newImplementation.
2. The vesting owner(multisig) needs to call the setImplementation() with the same implementation contract address to modify the implementation/logic.